lib/instrumentation/modules/memcached.js

```diff
@@ -0,0 +1,48 @@
+'use strict'
+
+var semver = require('semver')
+
+var shimmer = require('../shimmer')
+
+module.exports = function (memcached, agent, { version, enabled }) {
+  if (!enabled) {
+    return memcached
+  }
+  if (!semver.satisfies(version, '>=2.2.0')) {
+    agent.logger.debug('Memcached version %s not supported - aborting...', version)
+    return memcached
+  }
+  agent.logger.debug('shimming memcached.prototype.command')
+  shimmer.wrap(memcached.prototype, 'command', wrapCommand)
+  return memcached
+
+  // Wrap the generic command that is used to build touch, get, gets etc
+  function wrapCommand (original) {
+    return function wrappedCommand () {
+      if (typeof arguments[0] === 'function') {
+        var query = arguments[0]()
+        // If the callback is not a function the user doesn't care about result
+        if (query && typeof query.callback === 'function') {
+          var span = agent.startSpan(`memcached.${query.type}`, `db.memcached.${query.type}`)
+          if (span) {
+            span.setDbContext({ statement: `${query.type} ${query.key}`, type: 'memcached' })
+            agent.logger.debug('intercepted call to memcached.prototype.command %o', { id: span.id, type: query.type })
+            query.callback = wrapCallback(query.callback, span)
+            // Rewrite the query compiler with the wrapped callback
+            arguments[0] = function queryCompiler () {
+              return query
+            }
+          }
+        }
+      }
+      return original.apply(this, arguments)
+
+      function wrapCallback (cb, span) {
+        return function wrappedCallback () {
```

Comment:
As we don't use this more than once in this patch, if you wanted to simplify the code you could just move this function up inline, and just set it directly on the `query.callback` line. But now we're just into code cleanup territory.

Reply:
It starts to become a little hard to read then because we'd need to write the original function to a placeholder and then call that placeholder as the callback.